### PR TITLE
CNS-770: Pairing module readme

### PR DESCRIPTION
**Pull request by oren-lava**

### CODE REVIEW

File: x/pairing/README.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+
+Stake entries' storage is managed by the epochstorage module. For more details, see its README.
+
+#### Unstake
+
+A provider can unstake and retrieve their coins. When a provider unstakes, they are removed from the pairing list starting from the next epoch. After a specified number of blocks called `UnstakeHoldBlocks` (a parameter of the epochstorage module), the provider is eligible to receive their coins back.
+
+#### Freeze
+
+Freeze Mode enables the Provider to temporarily suspend their node's operation during maintenance to avoid bad Quality of Service (QoS). Freeze/Unfreeze is applied on the next Epoch. The Provider can initiate multiple Freeze actions, but only one Freeze-per-chain can be take effect at a time.
```
Comment (Yaroms) [inline x/pairing/README.md:91]: remove 
"but only one Freeze-per-chain can be take effect at a time"

File: x/pairing/README.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+### Payments
+
+The Lava network connects blockchain data providers with consumers who need access to that data. Once a pairing is established, a consumer can send a message to the provider to request a specific blockchain query. The provider then sends a relay request to the blockchain, retrieves the required information, and forwards it to the consumer. After completing the requested service, the provider sends a relay payment transaction to the Lava network to receive payment.
+
+Here is the workflow for Lava's payment mechanism:
+
+1. The provider sends a relay payment transaction, which is signed by the consumer.
+2. The reported CU (compute units) from the relay payment transaction is tracked.
+3. On a monthly basis, all the providers' tracked CU are reviewed, and payments are sent for their accumulated services.
```
Comment (Yaroms) [inline x/pairing/README.md:226]: this is not monthly basis but subscription base

File: x/pairing/README.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+
+#### CU Tracking
+
+The goal of CU tracking is to keep records of serviced CUs by the providers, in order to determine the amount of payment they should receive at the end of the month. When a relay payment transaction occurs, the number of CUs associated with each relay is counted and saved under the provider who initiated the transaction. At the end of each month, the CU tracker is reset for all providers.
```
Comment (Yaroms) [inline x/pairing/README.md:230]: by the providers to a specific subscription

File: x/pairing/README.md [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+
+#### Providers Payment
+
+Providers receive payments once a month. These payments are funded by the funds that consumers paid for their subscriptions. The amount of coins to be sent to each provider depends on the number of tracked CUs they had over the previous month. The payment is calculated using the following formula:
```
Comment (Yaroms) [inline x/pairing/README.md:234]: this is not accurate, they recieve payment whenever a subscription expires

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 5]
```diff
@@ -235,7 +235,17 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 		// update provider payment storage with complainer's CU
 		err = k.updateProviderPaymentStorageWithComplainerCU(ctx, relay.UnresponsiveProviders, logger, epochStart, relay.SpecId, relay.CuSum, servicersToPair, project.Index)
 		if err != nil {
-			utils.LogLavaEvent(ctx, logger, types.UnresponsiveProviderUnstakeFailedEventName, map[string]string{"err:": err.Error()}, "Error Unresponsive Providers could not unstake")
+			reportedProviders := ""
+			for _, p := range relay.UnresponsiveProviders {
+				reportedProviders += p.String() + ", "
```
Comment (Yaroms) [inline x/pairing/keeper/msg_server_relay_payment.go:240]: you can use string.join instead

File: x/pairing/README.md (hunk 1 above)
Comment (oren-lava) [reply x/pairing/README.md]: done. see https://github.com/lavanet/lava/commit/6de4aabf932a0ccf06e72626912782c8de252add

File: x/pairing/README.md (hunk 2 above)
Comment (oren-lava) [reply x/pairing/README.md]: done. see https://github.com/lavanet/lava/commit/6de4aabf932a0ccf06e72626912782c8de252add

File: x/pairing/README.md (hunk 3 above)
Comment (oren-lava) [reply x/pairing/README.md]: done. see https://github.com/lavanet/lava/commit/6de4aabf932a0ccf06e72626912782c8de252add

File: x/pairing/README.md (hunk 4 above)
Comment (oren-lava) [reply x/pairing/README.md:234]: done. see https://github.com/lavanet/lava/commit/6de4aabf932a0ccf06e72626912782c8de252add

File: x/pairing/keeper/msg_server_relay_payment.go (hunk 5 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go]: done. see https://github.com/lavanet/lava/commit/6de4aabf932a0ccf06e72626912782c8de252add

**Review by Yaroms — APPROVED** [review]

**Pull request merged**